build.gradle

```diff
@@ -0,0 +1,138 @@
+plugins {
+    id 'com.github.johnrengelman.shadow' version '8.1.1'
+    id 'java'
+}
+
+// Important for AutoPlugs Self-Updater!
+// Also take look at the generateAutoplugProperties task where these properties get turned into the actual autoplug.properties file.
+group = 'com.osiris.autoplug.client'
+version = '7.5.18'
+description = 'Responsible for all the main actions.'
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+// Important for AutoPlugs Self-Updater!
+// Generates a autoplug.properties file inside the generated jar file, with the stuff inside <properties>
+def generateAutoplugProperties = task('generateAutoplugProperties') {
+    def propertiesFile = file("$buildDir/resources/main/autoplug.properties")
+    outputs.file(propertiesFile)
+
+    doLast {
+        propertiesFile.text = """# Properties
+                                |# Generated on: ${new Date()}
+                                |java.version=${System.properties['java.version']}
+                                |version=${project.version}
+                                |main-class=com.osiris.autoplug.client.Main
+                                |slf4j.version=2.0.13
+                                |name=AutoPlug-Client
+                                |project.build.sourceEncoding=UTF-8
+                                |id=0
+                                |installation-path=./AutoPlug-Client.jar
+                                |""".stripMargin()
+    }
+}
+
+// Make this jar executable
+shadowJar {
+    archiveFileName = 'AutoPlug-Client.jar'
+    // Output into directory for manual tests
+    destinationDirectory = file("AP-TEST-SERVER")
+    dependsOn generateAutoplugProperties
+}
+
+// Configure jar task
+jar {
+    from sourceSets.main.output
+    archiveFileName = 'original-AutoPlug-Client.jar'
```

Comment:
I see `AutoPlug-Client.jar` is used a few times, maybe create it as string/property?

Reply:
Okay @Osiris-Team, I changed all of them to point to ``rootProject.name`` which is defined as ``AutoPlug-Client`` in the ``settings.gradle`` file.